java-advanced-training/src/main/java/com/epam/java_advanced_training/service/ProductService.java

```diff
@@ -0,0 +1,47 @@
+package com.epam.java_advanced_training.service;
+
+import com.epam.java_advanced_training.entity.Product;
+import com.epam.java_advanced_training.exception.ProductNotFoundException;
+import com.epam.java_advanced_training.repository.ProductRepository;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ProductService {
+
+  private final ProductRepository productRepository;
+
+  public Product createProduct(Product product) {
+    return productRepository.save(product);
+  }
+
+  public List<Product> getAllProducts() {
+    return productRepository.findAll();
+  }
+
+  public Product getProductById(Long id) {
+    return productRepository.findById(id)
+        .orElseThrow(() -> new ProductNotFoundException("Product with ID " + id + " not found"));
```

Comment:
It is business and case specific internally created exception that is under our management - so we can simplify interaction with it by putting string message ""Product with ID not found" inside the exception itself with ID as a parameter of exception constructor.